mambaSharedFramework/HLS Models/Playlist Structure/HLSPlaylistStructure.swift

```diff
@@ -372,16 +372,44 @@ fileprivate struct HLSPlaylistStructureConstructor {
         var currentSegmentDuration: CMTime = CMTime.invalid
         var discontinuity = false
         let tagDescriptor = self.tagDescriptor(forTags: tags)
-        
+
+        // collect indices for media sequence and skip tags as they impact the initial media sequence value
+        var mediaSequenceTagIndices = [Int]()
+        var skipTagIndices = [Int]()
+        tags.enumerated().forEach {
+            switch $0.element.tagDescriptor {
+            case PantosTag.EXT_X_MEDIA_SEQUENCE: mediaSequenceTagIndices.append($0.offset)
+            case PantosTag.EXT_X_SKIP: skipTagIndices.append($0.offset)
+            default: break
+            }
+        }
+
         // figure out our media sequence start (defaults to 1 if not specified)
-        let mediaSequenceTags = tags.filter{ $0.tagDescriptor == PantosTag.EXT_X_MEDIA_SEQUENCE }
-        if mediaSequenceTags.count > 0 {
-            assert(mediaSequenceTags.count == 1, "Unexpected to have more than one media sequence")
-            if let startMediaSequence: MediaSequence = mediaSequenceTags.first?.value(forValueIdentifier: PantosValue.sequence) {
+        if mediaSequenceTagIndices.count > 0 {
+            assert(mediaSequenceTagIndices.count == 1, "Unexpected to have more than one media sequence")
+            if
+                let mediaSequenceIndex = mediaSequenceTagIndices.first,
+                let startMediaSequence: MediaSequence = tags[mediaSequenceIndex].value(
+                    forValueIdentifier: PantosValue.sequence
+                )
+            {
                 currentMediaSequence = startMediaSequence
             }
         }
-        
+
+        // account for any skip tags (since a delta update replaces all segments earlier than the skip boundary, the
+        // SKIPPED-SEGMENTS value will effectively update the current media sequence value of the first segment, so safe
+        // to do this here and not within the looping through media group tags below).
+        if skipTagIndices.count > 0 {
```

Comment:
This approach seems to be more consistent with how the rest of the project is structured but I do wonder how much we care if there are multiple skip tags in the playlist? It should never happen so I guess you would catch it in development?

Not something that needs too be addressed here but I guess the alternatives would be:
- just checking for most recent skip tag or
- failing the HLS parse/validation if there were multiple tags?

Reply:
Yes, I was trying to keep in line with what existed, but I think I see where you're coming from on this.

I've made an update to avoid unnecessary looping through all the tags in the playlist here: d7c4e21269820780d497c0fdb9992a31e5964cc6

What do you think about that? I split it off in its own commit so that it's easier to revert if you aren't comfortable with the change.